components/style/values/specified/mod.rs

```diff
@@ -339,47 +367,61 @@ impl ToComputedValue for Angle {
     }
 
     fn from_computed_value(computed: &Self::ComputedValue) -> Self {
-        Angle {
-            radians: computed.radians(),
-            was_calc: false,
-        }
+        Angle::from_radians(computed.radians())
     }
 }
 
 impl Angle {
+    /// Returns an angle with the given value in degrees.
+    pub fn from_degrees(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Degree, was_calc: false }
+    }
+    /// Returns an angle with the given value in gradians.
+    pub fn from_gradians(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Gradian, was_calc: false }
+    }
+    /// Returns an angle with the given value in turns.
+    pub fn from_turns(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Turn, was_calc: false }
+    }
+    /// Returns an angle with the given value in radians.
+    pub fn from_radians(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Radian, was_calc: false }
```

Comment:
The only way to access the value of angle is through `radians` and it is checked there.

It may be a problem if NaN is passed to Angle::from_{radians, degrees, ...} and the angle is serialized with ToCss. Is this problematic or not?